### PR TITLE
Refactor.

### PR DESCRIPTION
**Pull request by batuhansk**
This pull request includes refactors applied throughout the library. Also, there are typo and style fixes. It looks better than ever.

### CODE REVIEW

**Comment by emrcftci** [comment]
Could you please rebase the master to solve conflicts?

File: UILibraries/ImageSlider/ImageSlider/Classes/Core/ImageSliderCollectionViewCell/ImageSliderCollectionViewCell.swift [hunk 1]
```diff
@@ -15,7 +15,7 @@ protocol ImageSliderCollectionViewCellInterface: class {
 }
 
 final class ImageSliderCollectionViewCell: UICollectionViewCell {
-    lazy var contentImageView: PinchableImageView = {
+    private (set) lazy var contentImageView: PinchableImageView = {
```
Comment (emrcftci) [inline UILibraries/ImageSlider/ImageSlider/Classes/Core/ImageSliderCollectionViewCell/ImageSliderCollectionViewCell.swift:18]: you can remove the unnecessary space between `private` and `(set)` -> `private(set)`.

**Comment by batuhansk** [comment]
I will take a look when I am available.

Thanks for reminding.

File: UILibraries/ImageSlider/ImageSlider/Classes/Core/ImageSliderView.swift [hunk 2]
```diff
@@ -45,9 +45,7 @@ public final class ImageSliderView: UIView {
         return collection
     }()
     
-    private lazy var pager: ImageSliderPagerView = {
-        return ImageSliderPagerView(frame: .zero)
-    }()
+    private lazy var pager = ImageSliderPagerView(frame: .zero)
```
Comment (emrcftci) [inline UILibraries/ImageSlider/ImageSlider/Classes/Core/ImageSliderView.swift:48]: simply just use `ImageSliderPagerView()` in this way the frame will be `zero`.

File: UILibraries/ImageSlider/ImageSlider/Classes/Core/ImageSliderCollectionViewCell/ImageSliderCollectionViewCell.swift (hunk 1 above)
Comment (batuhansk) [reply UILibraries/ImageSlider/ImageSlider/Classes/Core/ImageSliderCollectionViewCell/ImageSliderCollectionViewCell.swift:18]: There is no strict rule about that. It's just about the code styling and looks better with a space to me. 😊

File: UILibraries/ImageSlider/ImageSlider/Classes/Core/ImageSliderView.swift [hunk 3]
```diff
@@ -116,7 +114,8 @@ extension ImageSliderView: ImageSliderViewInterface {
     }
 
     public func scroll(to xOffset: Double, animate: Bool) {
-        collectionView.setContentOffset(CGPoint(x: CGFloat(xOffset), y: collectionView.contentOffset.y), animated: animate)
+        let offset = CGPoint(x: CGFloat(xOffset), y: collectionView.contentOffset.y)
+        collectionView.setContentOffset(offset, animated: animate)
```
Comment (emrcftci) [inline UILibraries/ImageSlider/ImageSlider/Classes/Core/ImageSliderView.swift:118]: there is a typo `animate` to `animated`.
**EDIT:** I see, the usage of `animate` like this but I think it should be like `animated`.

File: UILibraries/ImageSlider/ImageSlider/Classes/Core/ImageSliderView.swift (hunk 2 above)
Comment (batuhansk) [reply UILibraries/ImageSlider/ImageSlider/Classes/Core/ImageSliderView.swift:48]: Sure, it's just an existing code.

File: Utils/Utils/Common/Helpers/BlockTapGesture.swift [hunk 4]
```diff
@@ -15,15 +15,15 @@ class BlockTapGesture: UITapGestureRecognizer {
         super.init(target: target, action: action)
     }
 
-    public convenience init (tapCount: Int = 1, fingerCount: Int = 1, action: ((UITapGestureRecognizer) -> Void)?) {
+    public convenience init(tapCount: Int = 1, fingerCount: Int = 1, action: ((UITapGestureRecognizer) -> Void)?) {
```
Comment (emrcftci) [inline Utils/Utils/Common/Helpers/BlockTapGesture.swift:18]: 💯

File: Utils/Utils/Extensions/Encodable+Extension.swift [hunk 5]
```diff
@@ -21,7 +21,7 @@ public extension Encodable {
             encoder.outputFormatting = .sortedKeys
         }
         guard let data = try? encoder.encode(self),
-        let jsonString  = String(data: data, encoding: .utf8) else { return nil }
+        let jsonString = String(data: data, encoding: .utf8) else { return nil }
```
Comment (emrcftci) [inline Utils/Utils/Extensions/Encodable+Extension.swift:24]: indentation is wrong in here.

File: Utils/Utils/Extensions/UIColor+Extension.swift [hunk 6]
```diff
@@ -10,22 +10,26 @@ import UIKit
 
 public extension UIColor {
     var hexString: String? {
-        if let components = self.cgColor.components {
-            let red = components[0]
-            let green = components[1]
-            let blue = components[2]
-            return  String(format: "%02X%02X%02X", (Int)(red * 255), (Int)(green * 255), (Int)(blue * 255))
+        guard let components = self.cgColor.components, components.count == 3 else {
```
Comment (emrcftci) [inline Utils/Utils/Extensions/UIColor+Extension.swift:13]: please remove the unnecessary `self` before `cgColor`.

**Pull request closed without merging**